pyccel/ast/core.py

```diff
@@ -3331,6 +3332,22 @@ def attributes(self):
         """
         return self._attributes
 
+    @property
+    def is_deallocated(self):
+        """
+        Indicates whether the class has been deallocated.
+
+        Returns True if the class instance has been deallocated, False otherwise.
+        """
+
+        return self._is_deallocated
+
+    @is_deallocated.setter
+    def is_deallocated(self, value):
+        if not isinstance(value, bool):
+            raise TypeError("is_deallocated must be a boolean (Bool)")
+        self._is_deallocated = value
```

Comment:
I expected this variable to appear in the generated code not the Pyccel code. How do you handle if blocks?

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1508/commits/23157f9477e3bb5c71b413760b2855a1d08bcb5d.